torchgeo/transforms/temporal.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo temporal transforms."""
+
+from typing import Any
+
+from einops import rearrange
+from kornia.augmentation._3d.geometric.base import GeometricAugmentationBase3D
+from torch import Tensor
+
+
+class Rearrange(GeometricAugmentationBase3D):
+    """Rearrange tensor between [B, T, C, H, W] and [B, T*C, H, W]."""
```

Comment:
```suggestion
    """Rearrange tensor dimensions.
        
    Examples:
        To insert a time dimension::
        
            Rearrange('b (t c) h w -> b t c h w', c=1)
           
        To collapse the time dimension::
        
            Rearrange('b t c h w -> b (t c) h w')
    """
```
This is both:

* more generic (we can rearrange any dimensions, not just time)
* more specific (here are examples of how to use it)

Let's see how the formatting looks.